player/score_bounded_mcts/score_bounded_mcts_node.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+from game_logic.game_status import GameStatus
+from game_logic.igame_state import IGameState
+from player.mcts.mcts_node import MctsNode
+from player.mcts.imcts_node import IMctsNode
+
+
+class ScoreBoundedMctsNode(MctsNode):
+    def __init__(self, game_state: IGameState, action: int, parent: IMctsNode):
+        super().__init__(game_state, action, parent)
+        self._score_lower_bound = -np.inf
+        self._score_upper_bound = np.inf
+
+    def select(self, node, state):
+        while not node.untried_actions and node.game_state.available_moves:
+            node = node.select_child(self.game_state.is_first_player_move)
```

Comment:
I guess here you intend to select child depending on whether current node is first player. If so you should use `node.game_state.is_first_player_move`. All methods related to building tree are called directly on the root of the tree, so `self` relates to the root of the tree.

Reply:
Then you probably also don't need to override select method

Reply:
right, changed it to node. But I need to override select method to add subtrees pruning part

Reply:
Do you? I mean `select_child` is called on node. So you have access to node in `select_child` through `self`.